apps/studio/data/api-keys/[id]/api-key-id-update-mutation.ts

```diff
@@ -0,0 +1,70 @@
+import { useMutation, UseMutationOptions, useQueryClient } from '@tanstack/react-query'
+import { handleError, put } from 'data/fetchers'
+import { toast } from 'sonner'
+import { ResponseError } from 'types'
+import { apiKeysKeys } from '../keys'
+
+export interface UpdateAPIKeybyIdVariables {
+  projectRef?: string
+  id?: string
+  description?: string
+}
+
+export async function updateAPIKeysById(
+  { projectRef, id, description }: UpdateAPIKeybyIdVariables,
+  signal?: AbortSignal
+) {
+  if (typeof projectRef === 'undefined') throw new Error('projectRef is required')
+  if (typeof id === 'undefined') throw new Error('API key ID is required')
+
+  // @ts-ignore Just a sample here, TS lint will validate if the endpoint is valid
+  const { data, error } = await put('/v1/projects/{ref}/api-keys/{id}', {
```

Comment:
this isn't being used atm but API's method is PATCH right? not PUT, hence why the TS is complaining here
https://github.com/supabase/infrastructure/blob/develop/api/src/routes/v1/projects/ref/api-keys.controller.ts#L262

Reply:
it must had changed

Reply:
fixed